pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/db/dao/kit/BSPKitDao.java

```diff
@@ -101,18 +102,15 @@ public void setKitReceivedAndTriggerDDP(String kitLabel, boolean triggerDDP, BSP
                 stmt.setString(3, kitLabel);
                 int result = stmt.executeUpdate();
                 if (result > 1) { // 1 row or 0 row updated is perfect
-                    throw new RuntimeException("Error updating kit w/label " + kitLabel + " (was updating " + result + " rows)");
-                }
-                if (result == 1) {
-                    firstTimeReceived = true;
-                } else {
-                    firstTimeReceived = false;
+                    throw new DsmInternalError("Error updating kit w/label " + kitLabel + " (updated " + result + " rows)");
                 }
+                firstTimeReceived = result == 1;
             } catch (Exception e) {
                 logger.error("Failed to set kit w/ label " + kitLabel + " as received ", e);
             }
             if (triggerDDP) {
                 BSPKit bspKit = new BSPKit();
+                // TODO: this should not be in a DB transaction since it makes a call to an external service -DC
```

Comment:
Ah, I see it, under the covers this is making an http call to DSS.  Agreed that it would be best to not hold a transaction during an external call like this.